lib/pubsub/topic.js

```diff
@@ -96,6 +102,37 @@ Topic.formatName_ = function(projectId, name) {
 };
 
 /**
+ * Wrapper for makeReq_ that automatically attempts to create a topic if it
+ * does not yet exist.
+ *
+ * @private
+ */
+Topic.prototype.autoCreateWrapper_ = function(method, path, q, body, callback) {
+  var self = this;
+
+  var createAndRetry = function(){
+    self.pubsub.createTopic(self.unformattedName, function(err) {
+      if (err) {
+        callback(err);
+        return;
+      }
+      self.origMakeReq_(method, path, q, body, callback);
+    });
+  };
+
+  this.origMakeReq_(method, path, q, body, function(err, res) {
+    if (method !== 'DELETE' && err && err.code === 404) {
```

Comment:
ping?
If we changed the if clause to:
```javascript
if (err && err.code === 404 && method !== 'DELETE') {
}
```
We can remove one comparison in normal cases. gcloud-node already has lots of abstraction and I feel it's slower than expected (see https://github.com/GoogleCloudPlatform/gcloud-node/issues/441), this single change might not contribute much in terms of performance, but it is important to always keep performance-wise implementation in mind, otherwise we'll end up having such small inefficiency everywhere and it will get slower and slower.